[docs] add known issue where static roles get rotated on upgrade (#30145)
hashicorp/vault: website/content/partials/known-issues/static-role-premature-rotations.mdx

### Static role rotations on upgrade ((#static-role-rotations))

#### Affected Versions
- 1.19.0, 1.18.5, 1.17.12, 1.16.16

#### Issue

> LDAP or Database static-role credentials created before the upgrade are rotated once automatically during the upgrade. 
> After this one-time rotation, the static roles behave as expected.

Reviewer: ```suggestion
Vault automatically rotates existing static roles tied to database and LDAP
credentials once when upgrading to an affected version. After the one-time
rotation, the static roles behave as expected.
```
Style correction: write in active voice

#### Workaround

> Avoid upgrading to the affected versions if you rely on LDAP or Database static roles. A fix is currently 
> in progress and will be included in an upcoming release.

Reviewer: ```suggestion
If you rely on LDAP or static database roles, avoid upgrading to the affected
versions until we fix the issue.
```
Style correction: write in active voice, avoid speaking to future releases
